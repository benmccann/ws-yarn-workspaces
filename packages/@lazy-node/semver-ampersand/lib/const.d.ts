export declare const reAmpersandAndSpaces: RegExp;
export declare const reInvalidCharacter: RegExp;
