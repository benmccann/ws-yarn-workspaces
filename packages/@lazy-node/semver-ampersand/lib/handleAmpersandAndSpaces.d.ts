import { IOptions } from './types';
export declare function handleAmpersandAndSpaces<T>(versionRange: T, options?: IOptions): T;
export default handleAmpersandAndSpaces;
