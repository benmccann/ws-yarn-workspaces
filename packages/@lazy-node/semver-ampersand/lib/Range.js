"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Range = void 0;
const range_1 = __importDefault(require("semver/classes/range"));
const handleVersionRange_1 = require("./handleVersionRange");
class Range extends range_1.default {
    constructor(rawRange, optionsOrLoose) {
        const range = handleVersionRange_1.handleVersionRange(rawRange);
        super(range, optionsOrLoose);
        if (typeof rawRange === 'string' && range !== rawRange) {
            this.rawRange = rawRange;
        }
    }
}
exports.Range = Range;
//# sourceMappingURL=Range.js.map