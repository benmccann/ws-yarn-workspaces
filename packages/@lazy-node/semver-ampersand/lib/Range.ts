import semverRange from 'semver/classes/range';
import { Options } from 'semver';
import { handleVersionRange } from './handleVersionRange';

export class SemverRange extends semverRange
{
	rawSource?: string | Range;

	constructor(rawSource: string | SemverRange | semverRange, optionsOrLoose?: boolean | Options) {
		const range = handleVersionRange(rawSource);

		super(range, optionsOrLoose);

		if (typeof rawSource === 'string' && range !== rawSource)
		{
			this.rawSource = rawSource;
		}
	}

}

export { SemverRange as Range }

export default SemverRange
