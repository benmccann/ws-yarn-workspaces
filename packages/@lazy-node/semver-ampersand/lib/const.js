"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumSemverVersion = exports.separatorDoubleVerticalBar = exports.reDoubleVerticalBar = exports.reInvalidCharacterOrAmpersand = exports.reInvalidCharacter = exports.reAmpersandAndSpaces = exports.reSpaces = void 0;
exports.reSpaces = /\s+/g;
exports.reAmpersandAndSpaces = /[&\s]+/g;
exports.reInvalidCharacter = /[^\s\w.*\-><~!^|=+]/;
exports.reInvalidCharacterOrAmpersand = /[^\s\w.*\-><~!^|=+&]/;
exports.reDoubleVerticalBar = /\s*\|\|\s*/g;
exports.separatorDoubleVerticalBar = ' || ';
var EnumSemverVersion;
(function (EnumSemverVersion) {
    /**
     * nothing is allowed
     */
    EnumSemverVersion["NULL"] = "<0.0.0-0";
    /**
     * nothing is forbidden
     */
    EnumSemverVersion["STAR"] = "*";
    EnumSemverVersion["ANY"] = "";
})(EnumSemverVersion = exports.EnumSemverVersion || (exports.EnumSemverVersion = {}));
//# sourceMappingURL=const.js.map