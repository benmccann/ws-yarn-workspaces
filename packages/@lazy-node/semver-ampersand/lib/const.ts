
export const reSpaces = /\s+/g;

export const reAmpersandAndSpaces = /[&\s]+/g;

export const reInvalidCharacter = /[^\s\w.*\-><~!^|=+]/;

export const reInvalidCharacterOrAmpersand = /[^\s\w.*\-><~!^|=+&]/;

export const reDoubleVerticalBar = /\s*\|\|\s*/g;

export const separatorDoubleVerticalBar = ' || ' as const;
