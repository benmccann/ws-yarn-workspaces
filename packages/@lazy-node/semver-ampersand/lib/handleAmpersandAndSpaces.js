"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleAmpersandAndSpaces = handleAmpersandAndSpaces;
const const_1 = require("./const");
function handleAmpersandAndSpaces(versionRange, options) {
    if (typeof versionRange === 'string') {
        return versionRange.replace((options === null || options === void 0 ? void 0 : options.noAmpersand) ? const_1.reSpaces : const_1.reAmpersandAndSpaces, ' ').trim();
    }
    return versionRange;
}
exports.default = handleAmpersandAndSpaces;
//# sourceMappingURL=handleAmpersandAndSpaces.js.map