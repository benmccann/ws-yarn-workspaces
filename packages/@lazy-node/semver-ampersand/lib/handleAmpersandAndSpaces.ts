import { reAmpersandAndSpaces } from './const';

export function handleAmpersandAndSpaces<T>(versionRange: T): T
{
	if (typeof versionRange === 'string')
	{
		return versionRange.replace(reAmpersandAndSpaces, ' ').trim() as any
	}

	return versionRange
}

export default handleAmpersandAndSpaces
