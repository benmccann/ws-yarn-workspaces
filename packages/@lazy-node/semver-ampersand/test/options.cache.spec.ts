import { Range } from '../lib/Range';

test(`check cache is work`, () =>
{
	let range = '>01.02.03';

	expect(new Range(range, true)).toMatchSnapshot();
	//expect(actual).toBeInstanceOf(Date);
	expect(() => new Range(range)).toThrowErrorMatchingSnapshot();

});


