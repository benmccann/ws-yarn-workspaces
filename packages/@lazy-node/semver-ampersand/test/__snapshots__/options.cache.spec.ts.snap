// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check cache is work 1`] = `
SemverRange {
  "includePrerelease": false,
  "loose": true,
  "options": Object {
    "includePrerelease": false,
    "loose": true,
  },
  "range": ">1.2.3",
  "raw": ">01.02.03",
  "set": Array [
    Array [
      Comparator {
        "loose": true,
        "operator": ">",
        "options": Object {
          "loose": true,
        },
        "semver": SemVer {
          "build": Array [],
          "includePrerelease": false,
          "loose": true,
          "major": 1,
          "minor": 2,
          "options": Object {
            "loose": true,
          },
          "patch": 3,
          "prerelease": Array [],
          "raw": "01.02.03",
          "version": "1.2.3",
        },
        "value": ">1.2.3",
      },
    ],
  ],
}
`;

exports[`check cache is work 2`] = `"Invalid comparator: >01.02.03"`;
